### DISCUSSION TITLE
"Zoom In" for selected marker in Clustered markerSet

### DISCUSSION THREAD

**Discussion by kmurari103** [iTwin/itwinjs-core#3824] [Q&A]
Hi there,
I am facing some difficulty when i try to zoom to the selected marker in clustered marker set. Actually the list of marker is displayed as a table and on clicking specific row, i call `zoom()`  method to zoom to that location, but zoom is not enough to point to that selected marker . the entire cluster of marker is displayed with a number inside it. When i click on a marker which is not part of any cluster then zoom takes me to that selected marker in the viewport. Is there a way to zoom to selected marker even if that marker is part of some clustered set of markers?

Here is how i'm trying to achieve this-
`IModelApp.viewManager.selectedView!.zoom(row.original.worldLocation, 1, { animateFrustumChange: true });`
![image](https://user-images.githubusercontent.com/27068008/174061961-c98f6b1b-9994-49d6-a989-3f62bf79a732.png)

**Comment by bbastings** [comment]
You aren't actually zooming in since you've specified a factor of 1, you are just centering the view around the specified point.

Clustering is based on the marker's having their view rect overlap, that overlap will be largely unaffected by just a change to the center of the view, so the markers will continue to cluster.

You'd have to do something with your marker set to stop the clustering, like maybe hiding all the other markers in this scenario, etc.

**Reply by pmconne** [reply]
If you do want to zoom to the marker consider using [Viewport.zoomToVolume](https://www.itwinjs.org/reference/core-frontend/views/viewport/zoomtovolume/). You'll have to supply a volume that roughly encloses your marker. One way to get that is to pass the corners of the marker's view rect to [Viewport.viewToWorld](https://www.itwinjs.org/reference/core-frontend/views/viewport/viewtoworld/).
